source/isaaclab/isaaclab/utils/buffers/circular_buffer.py

```diff
@@ -100,6 +104,8 @@ def reset(self, batch_ids: Sequence[int] | None = None):
             batch_ids = slice(None)
         # reset the number of pushes for the specified batch indices
         self._num_pushes[batch_ids] = 0
+        # re-initialization is required
+        self._all_initialized = False
```

Comment:
NIT: Instead of "all_initialized", a better name might be "_has_reset"?

Syntax-wise, initialization sounds more like creation of the buffers, while here it is more resetting them.